master/internal/model/authz_basic_impl.go

```diff
@@ -46,10 +46,22 @@ func (a *ModelAuthZBasic) CanCreateModel(ctx context.Context,
 func (a *ModelAuthZBasic) CanDeleteModel(ctx context.Context, curUser model.User,
 	m *modelv1.Model, workspaceID int32,
 ) error {
-	// TODO: Modify model UserID to use UserID
 	curUserIsOwner := m.UserId == int32(curUser.ID)
 	if !curUser.Admin && !curUserIsOwner {
-		return fmt.Errorf("non admin users may not delete another user's models")
```

Comment:
this probably should have returned `PermissionDeniedError` wdy think?

Reply:
Agreed! I'll take a quick look through the model's authz files and make sure we're using the `PermissionDeniedError`

Reply:
@hamidzr Actually looking at this does it make sense to use `PermissionDeniedError` here, since it expects a list of expected perms?